atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/qos/QosCassandraClient.java

```diff
@@ -67,18 +67,21 @@ public QosCassandraClient(CassandraClient client, QosClient qosClient) {
     public Map<ByteBuffer, List<ColumnOrSuperColumn>> multiget_slice(String kvsMethodName, TableReference tableRef,
             List<ByteBuffer> keys, SlicePredicate predicate, ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
+        int numberOfQueriedRows = keys.size();
+
         return qosClient.executeRead(
                 () -> client.multiget_slice(kvsMethodName, tableRef, keys, predicate, consistency_level),
-                ThriftQueryWeighers.MULTIGET_SLICE);
+                ThriftQueryWeighers.multigetSlice(numberOfQueriedRows));
     }
 
     @Override
     public List<KeySlice> get_range_slices(String kvsMethodName, TableReference tableRef, SlicePredicate predicate,
             KeyRange range, ConsistencyLevel consistency_level)
             throws InvalidRequestException, UnavailableException, TimedOutException, TException {
+        int numberOfQueriedRows = range.count;
```

Comment:
this might be a little too aggressive, not sure how often we actually hit the limit

Reply:
some of my concern here is driven by the fact that we don't adjust the rate limiter if our estimate is too high - maybe we should just fix that